packages/lib-user/src/components/Contributors/helpers/handleGenerateExport.js

```diff
@@ -0,0 +1,102 @@
+import asyncStates from '@zooniverse/async-states'
+import { arrayOf, func, number, shape, string } from 'prop-types'
+
+import { getAllUsers } from './getAllUsers'
+
+async function generateExportWithWorker({ group, projects, stats, users }) {
+  return new Promise((resolve, reject) => {
+    const worker = new Worker(new URL('./generateExport.worker.js', import.meta.url))
```

Comment:
Creating the CSV was causing the UI to appear stuck at 98%. All the API requests had resolved, and the CSV generation started, which would cause the main browser JavaScript thread to hang. The only solution I could find was to put the CSV content creation on a separate thread with a [Web Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API). I'm not super familiar with Web Workers, largely followed the examples linked in the MDN docs.